src/main/java/uk/gov/ons/ctp/integration/mock/endpoints/common/RetrieveJson.java

```diff
@@ -0,0 +1,135 @@
+      throws IOException, CTPException {
+
+    String baseFileName = CaptureCache.normaliseFileName(name);
+
+    log.info(baseFileName);
+    HttpStatus responseStatus = HttpStatus.OK;
+    Object response = null;
+    String responseText = CaptureCache.readCapturedAiResponse(requestType, baseFileName);
+
+    if (responseText != null) {
+      // Convert captured AI response to an object, and return the target subset of data
+      if (requestType.getResponseClass().equals(String.class)) {
+        // Don't push it through Jackson, as it is already in String format
+        response = responseText;
+      } else {
+        response =
+            new ObjectMapper().readerFor(requestType.getResponseClass()).readValue(responseText);
+      }
+
+      switch (requestType) {
+        case AI_RH_POSTCODE:
+          AddressIndexRhPostcodeResultsDTO rhPostcodes =
+              (AddressIndexRhPostcodeResultsDTO) response;
+          List<AddressIndexRhPostcodeAddressDTO> rhPostcodeAddresses =
+              (List<AddressIndexRhPostcodeAddressDTO>)
+                  subset(rhPostcodes.getResponse().getAddresses(), offset, limit);
+          rhPostcodes.getResponse().setAddresses(rhPostcodeAddresses);
+          rhPostcodes.getResponse().setOffset(offset);
+          rhPostcodes.getResponse().setLimit(limit);
+          // Replicate the counting down of the confidence score
+          int confidence = 100000 + rhPostcodeAddresses.size();
+          for (AddressIndexRhPostcodeAddressDTO address : rhPostcodeAddresses) {
+            address.setConfidenceScore(confidence--);
+          }
+          break;
+        case AI_PARTIAL:
+          AddressIndexPartialResultsDTO partial = (AddressIndexPartialResultsDTO) response;
+          List<AddressIndexPartialAddressDTO> partialAddresses =
+              (List<AddressIndexPartialAddressDTO>)
+                  subset(partial.getResponse().getAddresses(), offset, limit);
+          partial.getResponse().setAddresses(partialAddresses);
+          partial.getResponse().setOffset(offset);
+          partial.getResponse().setLimit(limit);
+          break;
+        case AI_POSTCODE:
+          AddressIndexPostcodeResultsDTO postcodes = (AddressIndexPostcodeResultsDTO) response;
+          List<AddressIndexPostcodeAddressDTO> postcodeAddresses =
+              (List<AddressIndexPostcodeAddressDTO>)
+                  subset(postcodes.getResponse().getAddresses(), offset, limit);
+          postcodes.getResponse().setAddresses(postcodeAddresses);
+          postcodes.getResponse().setOffset(offset);
+          postcodes.getResponse().setLimit(limit);
+        case AI_EQ:
+          // Nothing to do for type-ahead response
+          break;
+        case AI_RH_UPRN:
+          // Nothing to do for uprn results
+          break;
+        case CASE_UPRN:
+          // Nothing to do for uprn results
+          break;
+        case CASE_ID:
+          // Nothing to do for uprn results
```

Comment:
Very minor, but you bunch all of the case lines together and just have a single break statement. Multiple inaccurate comments would also vanish

Reply:
done